lib/web_catalog/api_extractor.es6.js

```diff
@@ -0,0 +1,406 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+foam.CLASS({
+  name: 'apiExtractor',
+  package: 'com.web.catalog',
+  description: 'Extract api catalog from a object graph',
+  properties: [
+    {
+      name: 'defaults',
+      description: 'Default value of configuration',
+      factory: function() {
+        return {
+          // When reatainBuildInMembers is true,
+          // build-in properties of function and object
+          // are returned as APIs.
+          retainBuildInMembers: false,
```

Comment:
Done. Added properties and their class, documentation, name for retainBuildInMembers and retainConstantMembers